### ISSUE TITLE
Make torrequest (and other deps) optional

### ISSUE THREAD

**Issue by mjsir911** [sherlock-project/sherlock#2130]
## Checklist

- [x] I'm reporting a feature request
- [x] I've checked for similar feature requests including closed ones

## Description

Hi,

It would be cool if `sherlock.py` was modified a bit to not *require* the `torrequest` import even if I have no interest in downloading via tor, preferably deferring to importing when it's actually needed or maybe wrapping in an exception handler and/or stubbing out. 

Considering tor downloading won't work anyways without tor actually installed, shedding off an additional dependency (or two, I can't find `stem` being imported anywhere but it seems related) would be nice. The motivation behind this is `torrequest` and `stem` aren't currently packaged in gentoo's tree but all the other deps are.

In the same vein, `exrex` is only used within the tests but is pulled in the dependency list even for end-user installs. Moving this out to test-depends or something (I know python has this functionality, forget the name) would make sense.

**Comment by ppfeister** [comment]
torrequest wasn't available on Fedora / EPEL either. Seems like a dead project and I don't want to package something 6 years out of date. Here's my patch file that I'm using to remove it from my own build --

https://github.com/ppfeister/pkg/blob/master/sherlock/0001-Remove-tor.patch

Note that my patch is against the **new** package, #2111, which is pending code review

Removing features this way isn't ideal, but it's a fine short term fix in my opinion. I would like to figure something out for torrequest soon but that's my own fix for packaging in the meantime

I remember looking into removing stem but I don't remember why I didn't....... will have to investigate now. It may be a depend of another depend or something...

exrex has been removed from the dependency list in #2127, although it's very likely that I'll be adding it back for other tests. If I do, it'll be a dev depend and not in the default group. Should be gtg there either way.

**Comment by mjsir911** [comment]
> Removing features this way isn't ideal

Yeah, I'm interested in something much less destructive. Literally just moving the import would make things function until `--tor` is passed. Which is an acceptable tradeoff considering the app would fail anyways if tor isn't installed.

**Comment by ppfeister** [comment]
Raised a related poll #2192
